### DISCUSSION TITLE
transaction.from and transaction.to in the snap insight are lower case => Problem !

### DISCUSSION THREAD

**Discussion by javaspeak** [MetaMask/snaps#2059] [Q&A]
In the insight was reading transaction.from and transaction.to and relying on them to make calls to chain

I just discovered that they have been moved to lower case.  This means calls like eth_getLogs fail.

Do you have to put them to lower case.  In my opinion it is NOT helpful at all putting them to lower case.  Thought it was issues with block ranges and now find it is because transaction from the insight is lower case.

Why are you converting them to lower case?

**Comment by Montoya** [comment]
It is not our intention for the values to be lowercase, that happens upstream in the extension code and it's been that way for a long time. 

There's a couple ways to get around it. If you use `ethers`, it has a util `getAddress` that will normalize the address to a case-correct value. https://docs.ethers.org/v4/api-utils.html

If you want to do it manually, the steps are: 

1. Remove 0x prefix
2. sha3 the hash result from 1
3. Change nth letter of address according to the nth letter of the hash: 0,1,2,3,4,5,6,7 → Lowercase / 8,9,a,b,c,d,e,f → Uppercase

Let me know if that helps.
